src/client/client.h

```diff
@@ -548,7 +548,7 @@ class Client : public con::PeerHandler, public InventoryManager, public IGameDef
 	// Set of media filenames pushed by server at runtime
 	std::unordered_set<std::string> m_media_pushed_files;
 	// Pending downloads of dynamic media (key: token)
-	std::vector<std::pair<u32, std::unique_ptr<SingleMediaDownloader>>> m_pending_media_downloads;
```

Comment:
what the hell, you totally change the pointer behaviour. unique means it cannot be owned by anyone else, unlike shared_ptr

Reply:
It doesn't make a difference in this case but I agree it's not ideal.

Reply:
I read every single function that accesses m_pending_media_downloads and thought through them, I ran all the unit tests, and I tested connecting to the CTF server to verify that downloading media wasn't wrecked. It's quite possible I missed something - that's why we do reviews - but all I can say to "what the h*** you changed the behavior" is "yes, you bet I did".

Reply:
I considered a third option after looking at the code, which would be to re-order it so that the pointer doesn't need to be kept around after moving it to m_pending_media_downloads, but I am not familiar with whether this is threaded or not, and I did not think messing with the ordering would be wise. If that is a viable solution, I would want to hear it from someone who knows the code.

Reply:
It was suggested to me by a correspondent that I should try this, so I've tested it. There is no problem that I can tell, but I haven't gone through the code. Would you like me to commit it, and if so should I force push, or revert the last commit and add it on top?

Reply:
This is okay for me even if I suggested a different solution before.